Sources/StreamChat/Errors/ClientError.swift

```diff
@@ -19,10 +19,23 @@ public class ClientError: Error, CustomStringConvertible {
     /// An underlying error.
     public let underlyingError: Error?
     
-    var errorDescription: String? { underlyingError.map(String.init(describing:)) }
-    
     /// Retrieve the localized description for this error.
-    public var localizedDescription: String { message ?? errorDescription ?? "" }
+    public var localizedDescription: String {
+        if let message = message {
+            return message
+        }
+        
+        switch underlyingError {
+        case let clientError as ClientError:
+            return clientError.localizedDescription
+        case let errorPayload as ErrorPayload:
+            return errorPayload.message
+        case let engineError as WebSocketEngineError:
+            return engineError.reason
```

Comment:
Is this actually correct? 🤔 From the previous commits, I remember that the websocket engine error was an underlying error of `errorPayload`?

Reply:
Yeah, here is the recursive case:
```swift
case let clientError as ClientError:
            return clientError.localizedDescription
```

When ws is trying to connect and fails with connection error, the nested structure is:
`ClientError.ConnectionNotSuccessful -> ClientError.WebSocket -> WebSocketEngineError -> Error`

So in case of the WS error, as a localized description, we'll take `WebSocketEngineError.reason`